### ISSUE TITLE
[UWP] Viewer gets unusable slow with big Pdfs

### ISSUE THREAD

**Issue by leoek** [PSPDFKit/react-native#240]
# Details

The Viewer is really slow if a big pdf is opened. While scrolling the user has to wait for every page individually to load. Furthermore the first load of a page ist always blured for quite a while until it is displayed properly.

For Pdfs >500 pages the Viewer is not usable.

On IOS the Viewer is as fast as with pdfs which contain only a single page. Even the web version (https://web-examples.pspdfkit.com/custom) is much faster. 

Pdfs of this size are not unusual. The issue can be reproduced with this [Pdf (232 pages) ](https://investor-relations.lufthansagroup.com/fileadmin/downloads/de/finanzberichte/geschaeftsberichte/LH-GB-2017-d.pdf)

# Acceptance Criteria

- [ ] Performance on Windows is similar to the performance on IOS or at least faster than the pspdfkit for web version.

**Comment by steipete** [comment]
Just checking, did you test this with Windows 2.2 yet? The main focus of this release is performance: https://pspdfkit.com/blog/2019/pspdfkit-windows-2-2/

**Comment by leoek** [comment]
@steipete No, I tested it with version 2.0 (which was the newest I could get a few days ago). I will test with 2.2 next.

**Comment by steipete** [comment]
Much appreciated! Performance should be better than Web, since Web uses WebAssembly (which incurs a ~30% translation tax) vs UWP which runs the same code natively.

**Comment by leoek** [comment]
@steipete I just tested 2.2 and it is a little better. However it is still a lot slower than the web version in Chrome (but comparable to the performance in Edge).

Are you sure that the same code which runs with WebAssembly in Chrome is run natively in UWP?
And can you please merge #241? (I had to bump the version because the windows 2.2 merge had no version bump which resulted in yarn not resolving anything new)

**Comment by steipete** [comment]
UWP uses the EdgeHTML engine, so that's the browser you need to use for performance comparison. We plan to switch to Microsoft's WebView2 which is Chromium-based and offers a faster DOM and JavaScript- performance, once it is stable. Currently it still lacks some critical features including UWP support, but Microsoft is actively working on it.

https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview2

**Comment by nickwinder** [comment]
To add a little more to this. I just tested the Windows Catalog and React Native PSPDFKit Catalog side by side and the speeds are comparable from a human perspective.

**Comment by leoek** [comment]
@nickwinder What is the difference between "Windows Catalog and React Native PSPDFKit Catalog" I thought it was basically the same version.

@steipete I know that UWP uses Edge for rendering and if I compare it to Edge the performance seems similiar. However we got the Feedback from QA and Users that the experience on Windows with big pdfs is bad compared to IOS.

Do you know how far MS is with WebView2?

**Comment by nickwinder** [comment]
> @nickwinder What is the difference between "Windows Catalog and React Native PSPDFKit Catalog" I thought it was basically the same version.

Yes it does share the same backbone. I wanted to ensure confirm and reassure that the wrapper of react native was not causing any significant slow downs.

**Comment by steipete** [comment]
WebView2 is something we hope to migrate to in Q1 2020. The timeline really depends on how fast we get a beta that we could use to test.

**Comment by leoek** [comment]
@nickwinder that's correct, there seem to be no real slow downs because of the wrapper itself.

@steipete The microsoft docs seem like the beta of webview2 is already available: https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview2/gettingstarted

**Comment by steipete** [comment]
The beta is for WIN32 but not UWP yet.

**Comment by leoek** [comment]
> The beta is for WIN32 but not UWP yet.

Ah ok. I didn't notice that.

**Comment by nickwinder** [comment]
@leoek Just following up on this. There's still no WebView2 in sights, although there were more performance updates in 2.3.0 UWP. I'd love to know what stage we're at here.

**Comment by steipete** [comment]
We're pushing Microsoft for a preview, but it doesn't seem they can deliver something before January.
